custom/icds_reports/utils/aggregation_helpers/distributed/migration_forms.py

```diff
@@ -0,0 +1,87 @@
+from dateutil.relativedelta import relativedelta
+
+from custom.icds_reports.const import AGG_MIGRATION_TABLE
+from custom.icds_reports.utils.aggregation_helpers import month_formatter
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import (
+    StateBasedAggregationDistributedHelper,
+)
+
+
+class MigrationFormsAggregationDistributedHelper(StateBasedAggregationDistributedHelper):
+    helper_key = 'migration-forms'
+    ucr_data_source_id = 'static-migration_form'
+    aggregate_parent_table = AGG_MIGRATION_TABLE
+
+    def data_from_ucr_query(self):
+        month = self.month.replace(day=1)
+        current_month_start = month_formatter(self.month)
+        next_month_start = month_formatter(self.month + relativedelta(month=1))
+
+        query_params = {
+            "month": month_formatter(month),
+            "state_id": self.state_id,
+            "current_month_start": current_month_start,
+            "next_month_start": next_month_start,
+        }
+        return """
+        SELECT DISTINCT
+            %(state_id)s AS state_id,
+            supervisor_id,
+            %(month)s::DATE AS month,
+            person_case_id AS person_case_id,
+            LAST_VALUE(is_migrated) OVER w AS is_migrated,
+            CASE
```

Comment:
`migration_date` is the main thing on which the migration status behaviour is working.It basically used to identify in which month actually a person is migrated instead of marking migrated in all months. https://dimagi-dev.atlassian.net/browse/ICDS-1025 Check Nitish comment on January 16, 2020, 4:37 AM

Reply:
Right, I mean the `CASE` statement, when would `date_left` be blank and we have to fall back to submission date?

Reply:
@Rohit25negi

Reply:
Yeah, date_left is a question in the form which is **not required**. It may sometime come as empty because aww does not date when the people left or with other reasons. That's why we have to fall back to timeend